handle.c

```diff
@@ -111,7 +116,17 @@ int hdup(int h) {
 
 void *hquery(int h, const void *type) {
     CHECKHANDLE(h, NULL);
-    return hndl->vfs->query(hndl->vfs, type);
+    /* Try and use cached pointer first, otherwise do expensive virtual call.*/
```

Comment:
That's clever. Most handles expose only a single interface anyway, so there will be no virtual function calls except for the first one.